ArrayListAndLinkedList05/src/com/gmail/granovskiy/s/ArrayList.java

```diff
@@ -0,0 +1,61 @@
+package com.gmail.granovskiy.s;
+
+import java.util.Arrays;
+
+public class ArrayList<T> implements List<T> {
+    private int arrayLength = 7;
+    private Object[] arrayOne;
+    private int size = 0;
+
+    public ArrayList() {
+        arrayOne = new Object[arrayLength];
+    }
+
+    @Override
+    public void add(T t) {
+        if (size == arrayOne.length) {
+            resize(arrayOne.length + 1);
+            arrayOne[size++] = t;
+        } else {
+            arrayOne[size++] = t;
+        }
+    }
+
+    private void resize(int newArrayLength) {
+        arrayOne = Arrays.copyOf(arrayOne, newArrayLength);
```

Comment:
newArrayLength = arrayOne.length + 1
you will use method resize every iteration, it very slowly 
Arrays.copyOf(arrayOne, array.length * 2);